### ISSUE TITLE
npm start wants npm run first

### ISSUE THREAD

**Issue by omenking** [elthrasher/planetstack#1]
When following the readme it suggest to run `npm start` which then prompts for running `npm run deploy`

![image](https://user-images.githubusercontent.com/7776/129097601-63eb1973-1ec2-4f9f-b772-4176f1361935.png)

**Comment by omenking** [comment]
npm run deploy wants me to run cdk boostrap

![image](https://user-images.githubusercontent.com/7776/129097736-47799048-c3f2-4b49-a4ec-fc4968a5d5f8.png)

**Comment by omenking** [comment]
My CDK is out of date

> This CDK CLI is not compatible with the CDK library used by your application. Please upgrade the CLI to the latest version.
(Cloud assembly schema version mismatch: Maximum schema version supported is 9.0.0, but found 13.0.0)

Going to update my CDK

```
npm install -g aws-cdk@latest
```

**Comment by omenking** [comment]
![image](https://user-images.githubusercontent.com/7776/129099462-f4450531-b5b0-41b7-abea-2e75ef4ccae4.png)

> CloudWatch Logs role ARN must be set in account settings to enable logging

hmmm

**Comment by omenking** [comment]
https://aws.amazon.com/premiumsupport/knowledge-center/api-gateway-cloudwatch-logs/

Looks like the Role is part of the CloudFormation template that is generated so uncertain about its complaint.
 ```
  CWRole7EC94C1E:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: apigateway.amazonaws.com
        Version: "2012-10-17"
      ManagedPolicyArns:
        - Fn::Join:
            - ""
            - - "arn:"
              - Ref: AWS::Partition
              - :iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatch
 ```

![image](https://user-images.githubusercontent.com/7776/129102600-4bd526e5-6ba0-421e-862d-d7e64e2feae3.png)

**Comment by elthrasher** [comment]
Thanks for the issue @omenking.

Needing to `npm run deploy` before `npm start` was a design choice in the tooling since there isn't a good local dev experience for WebSocket API. Maybe it's not fully intuitive, but the idea is that you must deploy once to generate the WebSocket API url and then it gets stored in local config so you can run the UI locally, making changes and seeing the UI update while developing against the hosted API. Of course this isn't necessary if you just want to deploy the app.

It shouldn't be necessary to install `aws-cdk` globally. The bootstrap instructions might say that, but you could've probably also gotten away with `npx cdk bootstrap`.

As for the role, the account I'm testing in already has that cloudwatch role from another app and so that might be why it worked for me and not for you. The integration here between API Gateway and CloudWatch is pretty odd IMO in that a completely different API Gateway can basically set this up for you. I think what I left out of my code in this repo is a dependency to ensure that role is created FIRST. I don't see your entire Cfn output above, but I'm guessing the CW role hadn't been created yet when your service roles started failing? This might be the reason.

I'll try working through this to see if that conclusion is correct.

**Comment by elthrasher** [comment]
@omenking Not sure if I'm missing something here, but the CW Role thing seems to be a bit of a mess for apigwv2.

1. I deleted all stacks in my account and verified no roles were using `arn:aws:iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs`.
2. I edited my planetstack app to remove the role creation.
3. I deployed planetstack successfully!
4. I checked the APIGW console and noticed my old and now deleted CloudWatch role ARN.
5. I used the app and noticed no APIGW logs were being written.
6. I added back the role creation to planetstack and deployed.
7. I used the app and noticed logs were now being written and the new role ARN was showing in the console.

It looks to me like that CloudWatch role ARN gets cached for validation by APIGW, so I'm not able to even reproduce this validation issue in my account.

I went back to the APIGW (v1) code I took the role creation from and noticed there's a dependency that I didn't include in my code: https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/restapi.ts#L515

Can you check out [this branch](https://github.com/elthrasher/planetstack/pull/2) and see if it deploys? It could be I was just missing that dependency.

**Comment by omenking** [comment]
Cool! I'll check that branch.

**Comment by omenking** [comment]
Changed branches, deleted template.yaml, ran synth and deploy.

Same error

![image](https://user-images.githubusercontent.com/7776/129215850-c2742fdd-d692-4cc4-b241-742d50d7732c.png)

I think I will need to deploy in a different AWS account of mine.

**Comment by elthrasher** [comment]
Bummer. I'll try a different account later on.

**Comment by elthrasher** [comment]
Okay, I was able to reproduce in a different account and saw the exact same thing. I was right about the Cfn dependency. What is odd is the [@aws-cdk/apigateway construct](https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-apigateway/lib/restapi.ts#L515) has the account resource depend on the api gateway. That seemed wrong, but I figured they knew what they were doing ;)

Perhaps not? I swapped it so the api gateway depends on the account resource and then my app deployed fine. Check the branch. I pushed an update and it should work! Thanks for testing it out @omenking.
